src/spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -1,327 +1,405 @@
+            if levels is None:
+                levels = [f"level{i}" for i in range(num_levels)]
+            else:
+                levels = list(levels)
+                assert len(levels) == num_levels
+        else:
+            raise ValueError("Keys for cases must str or tuple")
+
+
+        study_folder = Path(study_folder)
+        study_folder.mkdir(exist_ok=False, parents=True)
+
+        (study_folder / "datasets").mkdir()
+        (study_folder / "datasets/recordings").mkdir()
+        (study_folder / "datasets/gt_sortings").mkdir()
+        (study_folder / "sorters").mkdir()
+        (study_folder / "sortings").mkdir()
+        (study_folder / "sortings" / "run_logs").mkdir()
+        (study_folder / "metrics").mkdir()
+
+        for key, (rec, gt_sorting) in datasets.items():
+            assert "/" not in key
+            assert "\\" not in key
+
+            # rec are pickle
+            rec.dump_to_pickle(study_folder / f"datasets/recordings/{key}.pickle")
+
+            # sorting are pickle + saved as NumpyFolderSorting
+            gt_sorting.dump_to_pickle(study_folder / f"datasets/gt_sortings/{key}.pickle")
+            gt_sorting.save(format="numpy_folder", folder=study_folder / f"datasets/gt_sortings/{key}")
+        
+        
+        info = {}
+        info["levels"] = levels
+        (study_folder / "info.json").write_text(json.dumps(info, indent=4), encoding="utf8")
+
+        # (study_folder / "cases.jon").write_text(
+        #     json.dumps(cases, indent=4, cls=SIJsonEncoder),
+        #     encoding="utf8",
+        # )
+        # cases is dump to a pickle file, json is not possible because of tuple key
+        (study_folder / "cases.pickle").write_bytes(pickle.dumps(cases))
 
-    def __repr__(self):
-        t = "Ground truth study\n"
-        t += "  " + str(self.study_folder) + "\n"
-        t += "  recordings: {} {}\n".format(len(self.rec_names), self.rec_names)
-        if len(self.sorter_names):
-            t += "  sorters: {} {}\n".format(len(self.sorter_names), self.sorter_names)
+        return cls(study_folder)
 
-        return t
 
     def scan_folder(self):
-        self.rec_names = get_rec_names(self.study_folder)
-        # scan computed names
-        self.computed_names = list(iter_computed_names(self.study_folder))  # list of pair (rec_name, sorter_name)
-        self.sorter_names = np.unique([e for _, e in iter_computed_names(self.study_folder)]).tolist()
-        self._is_scanned = True
+        if not (self.folder / "datasets").exists():
+            raise ValueError(f"This is folder is not a GroundTruthStudy : {self.folder.absolute()}")
+
+        with open(self.folder / "info.json", "r") as f:
+            self.info = json.load(f)
+        
+        self.levels = self.info["levels"]
+
+        for rec_file in (self.folder / "datasets/recordings").glob("*.pickle"):
```

Comment:
```suggestion
        for rec_file in (self.folder / "datasets" / "recordings").glob("*.pickle"):
```
I think that ".../..." might fail on Windows